packages/gatsby-theme-docs/configure-theme.js

```diff
@@ -0,0 +1,25 @@
+const pkg = require('./package.json');
+
+function configureThemeWithAddOns(themeOptions = {}) {
+  const { addOnPlugins, ...coreThemeOptions } = themeOptions;
+
+  if (!addOnPlugins || !Array.isArray(addOnPlugins)) {
+    throw new Error(
+      'Missing or wrong format for "addOnPlugins" option. Expected a list of Gatsby theme add-on plugins.'
+    );
+  }
+
+  const coreThemeConfig = {
+    resolve: pkg.name,
+    options: {
+      ...coreThemeOptions,
+      themeAddOns: addOnPlugins.map((plugin) =>
+        typeof plugin === 'string' ? plugin : plugin.resolve
+      ),
+    },
+  };
+  // Flatten theme plugins, so that they are loaded by Gatsby as themes
+  return [coreThemeConfig, ...addOnPlugins];
+}
+
+exports.configureThemeWithAddOns = configureThemeWithAddOns;
```

Comment:
Nitpick: can we re-export this from the `index.js` root of the package as you recommended generally?  That would make the configuration slightly more concise and we can refactor the actual location if we want.

Reply:
The `index.js` only exports React components, or things needed in the UI.
I wanted to keep that separate from the nodejs part.
The export file can still be there, even if we move things around. We just need to re-export things properly.